packages/@react-native-windows/telemetry/src/utils/projectUtils.ts

```diff
@@ -0,0 +1,215 @@
+ * @returns Whether the project at the given path has a react-native.config.js file.
+ */
+export async function usesReactNativeConfig(
+  projectRoot: string,
+): Promise<boolean> {
+  return fileExists('./react-native.config.js', projectRoot);
+}
+
+/**
+ * Gets whether the project at the given path is using TypeScript.
+ * @param projectRoot The project root path to look in.
+ * @returns Whether the project at the given path is using TypeScript.
+ */
+export async function usesTypeScript(projectRoot: string): Promise<boolean> {
+  return fileExists('./tsconfig.json', projectRoot);
+}
+
+/**
+ * Get the list of keys in the object for which the value is defined.
+ * @param obj The object to search.
+ * @returns The list of keys.
+ */
+function getDefinedKeys(obj: Record<string, any>): string[] {
+  return Object.keys(obj).filter(
+    value => obj[value] !== undefined && obj[value] !== null && value,
+  );
+}
+
+/**
+ * Given a react-native CLI config, determine the language of the RNW project if possible.
+ * @param config Config passed from react-native CLI.
+ * @returns The language of the RNW project.
+ */
+function getRnwLang(
+  config: Record<string, any> | null | undefined,
+): 'cpp' | 'cs' | null {
+  if (config) {
+    for (const project of config.projects) {
+      if (project.directDependency) {
+        return project.projectLang;
+      }
+    }
+  }
+  return null;
+}
+
+/**
+ * Returns true if the item is a boolean with a value of true or a string with a value of 'true'.
+ * @param item The item to parse.
+ * @returns The boolean value.
+ */
+function parseBoolean(item: any): boolean {
+  if (typeof item === 'boolean') {
+    return item;
+  } else if (typeof item === 'string') {
+    return item.toLowerCase() === 'true';
+  }
+  return false;
+}
+
+/**
+ * Calculate the project telemetry info from a react-native CLI config.
+ * @param config Config passed from react-native CLI.
+ * @returns The calculated project info.
+ */
+export async function configToProjectInfo(
+  config: Record<string, any>,
+): Promise<AppProjectInfo | DependencyProjectInfo | null> {
+  try {
+    const id: string = getProjectId(
+      require(path.join(config.root, 'package.json')).name,
+    );
+    const platforms = getDefinedKeys(config.project);
+
+    if ('windows' in config.project && config.project.windows !== null) {
+      const rnwLang = config.project.windows.project?.projectLang;
+      const usesTS = await usesTypeScript(config.project.windows.folder);
+      const usesRNConfig = await usesReactNativeConfig(
+        config.project.windows.folder,
+      );
+      const jsEngine = parseBoolean(
+        config.project.windows.experimentalFeatures?.UseHermes,
+      )
+        ? 'Hermes'
+        : 'Chakra';
+      const rnwSource = parseBoolean(
+        config.project.windows.experimentalFeatures?.UseExperimentalNuget,
+      )
+        ? 'NuGet'
+        : 'Source';
+
+      const dependencies: DependencyProjectInfo[] = [];
```

Comment:
we should probably capture WinUI2xVersion - and possibly WindowsTargetPlatformVersion

Reply:
Tracking: https://github.com/microsoft/react-native-windows/issues/9165